PocketKit/Sources/PocketKit/Tags/AddTagsView.swift

```diff
@@ -63,52 +54,126 @@ struct AddTagsView: View {
+            Text(tag)
+                .style(.tag)
+                .padding(Constants.tagPadding)
+                .background(Rectangle().fill(Color(.ui.grey6)))
+                .cornerRadius(4)
+                .lineLimit(1)
+                .onTapGesture {
+                    viewModel.removeTag(with: tag)
+                }
+                .accessibilityIdentifier("tag")
+                .matchedGeometryEffect(id: tag, in: animation)
+        }
+        
+        func getRows(screenWidth: CGFloat) -> [[String]] {
+            var rows: [[String]] = []
+            var currentRow: [String] = []
             
-            if totalWidth > safeWidth {
-                totalWidth = (!currentRow.isEmpty || rows.isEmpty ? tagWidth : 0)
+            var totalWidth: CGFloat = 0
+            let safeWidth: CGFloat = screenWidth
+            let padding: CGFloat = Constants.tagPadding * 2 + Constants.tagsHorizontalSpacing * 2
+        
+            viewModel.tags.forEach { tag in
+                let attributes = Style.tag.textAttributes
+                let tagWidth: CGFloat = tag.size(withAttributes: attributes).width + padding
+
+                totalWidth += tagWidth
+                
+                if totalWidth > safeWidth {
+                    totalWidth = (!currentRow.isEmpty || rows.isEmpty ? tagWidth : 0)
+                    rows.append(currentRow)
+                    currentRow.removeAll()
+                    currentRow.append(tag)
+                } else {
+                    currentRow.append(tag)
+                }
+            }
+        
+            if !currentRow.isEmpty {
                 rows.append(currentRow)
                 currentRow.removeAll()
-                currentRow.append(tag)
-            } else {
-                currentRow.append(tag)
             }
+            return rows
         }
+    }
     
-        if !currentRow.isEmpty {
-            rows.append(currentRow)
-            currentRow.removeAll()
+    struct OtherTagsView: View {
+        @ObservedObject
+        var viewModel: AddTagsViewModel
+        
+        var body: some View {
+            if let otherTags = viewModel.allOtherTags(), !otherTags.isEmpty {
+                List {
+                    Section(header: Text("All Tags").style(.sectionHeader)) {
+                        ForEach(otherTags, id: \.self) { tag in
+                            HStack {
+                                Text(tag).style(.tag)
+                                Spacer()
+                            }
+                            .contentShape(Rectangle())
+                            .onTapGesture {
+                                _ = viewModel.addTag(with: tag)
+                            }
+                        }
+                    }
+                }
+                .listStyle(.plain)
+                .accessibilityIdentifier("all-tags")
+            } else {
+                // TODO: Empty State View
```

Comment:
Is this still "TODO"?

Reply:
yes, it'll be addressed when we start working on the pre-app store polish epic
https://getpocket.atlassian.net/browse/IN-779